app/lib/staypuft/network_query.rb

```diff
@@ -155,6 +155,7 @@ def interface_hash_for_host(subnet_type_name, host)
     def gateway_hash_for_host(host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PUBLIC_API, host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PXE, host) unless gateway_hash[:subnet]
+      gateway_hash = interface_hash_for_host(Staypuft::SubnetType::EXTERNAL, host) if Staypuft::Role.compute.any? { |r| r.name == host.hostgroup.name } && host.deployment.nova_networking?
```

Comment:
Nevermind the comments on compute roles -- I missed the scroll-around.

Lines 157 and 158 still need to be swapped here though.